### ISSUE TITLE
Workflow path defined differently than primary descriptor

### ISSUE THREAD

**Issue by hyunnaye** [dockstore/dockstore#5330]
**Describe the bug**
The `workflow_path` of a workflow should be the relative path of primary descriptor. A correct implementation of this is `full_workflow_path` where it returns the absolute path of the primary descriptor file. However, `workflow_path` in some WDL workflows are set to be `.dockstore.yml` and `Dockstore.cwl` for CWL files.

Example of WDL workflow: 
https://dockstore.org/workflows/github.com/broadinstitute/viral-pipelines/assemble_refbased:master?tab=info

Example of CWL workflow: 
https://dockstore.org/workflows/github.com/h3abionet/h3agatk:1.0.1?tab=info

**To Reproduce**
Visit the links above in an local environment and test out each workflows `workflow_path` and `full_workflow_path`.

**Expected behavior**
 `workflow_path` and `full_workflow_path` should point to the same file

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/DOCK-2311)
┆Issue Number: DOCK-2311
┆Sprint: 105 - Colorado
┆Issue Type: Story

**Comment by denis-yuen** [comment]
Ok, I think this may be a miscommunication. 

https://dockstore.org/api/workflows/path/workflow/github.com%2Fbroadinstitute%2Fviral-pipelines%2Fassemble_refbased/published?include=validations%2Cauthors&subclass=BIOWORKFLOW&versionName=master

```
mode | "DOCKSTORE_YML"
full_workflow_path | "github.com/broadinstitute/viral-pipelines/assemble_refbased"
workflow_path | "/.dockstore.yml"
```

https://dockstore.org/api/workflows/path/workflow/github.com%2Fh3abionet%2Fh3agatk/published?include=validations%2Cauthors&subclass=BIOWORKFLOW&versionName=1.0.1

```
mode | "FULL"
full_workflow_path | "github.com/h3abionet/h3agatk"
workflow_path | "/Dockstore.cwl"
```

https://dockstore.org/api/workflows/path/workflow/github.com%2Fiwc-workflows%2Fgeneric-variant-calling-wgs-pe%2Fmain/published?include=validations%2Cauthors&subclass=BIOWORKFLOW&versionName=main
```
mode | "DOCKSTORE_YML"
full_workflow_path | "github.com/iwc-workflows/generic-variant-calling-wgs-pe/main"
workflow_path | "/.dockstore.yml"
```

So this should not be the case

>  workflow_path and full_workflow_path should point to the same file

The `full_workflow_path` is essentially an ID for the workflow. The `workflow_path` is a file path in the repository to what we consider the primary descriptor (I know it's confusing how we overloaded the word path). 
The `workflow_path` is stored as the `.dockstore.yml` for workflows that have one and the path to the primary descriptor which is `Dockstore.cwl` in the case of legacy registered workflows. 

TLDR: I think this is as designed and a side-effect of having both workflows registered as github app workflows and legacy workflows.
